src/components/game/player/info/CityInfo.tsx

```diff
@@ -1,21 +1,33 @@
 import * as React from 'react'
 
 import { FlexContainer } from '../../../styled/FlexContainer'
-import { ExtendedCity } from '../../city/city.types'
+import { ExtendedCity, City } from '../../city/city.types'
 import { YieldInfo } from './YieldInfo'
+import { units } from '../../units/units'
+import { StyledP } from '../../../styled/StyledP'
+import { ProgressInfo } from './ProgressInfo'
+import { EMPTY_PROGRESS } from '../../../../utils/utils'
 
-interface TileInfoProps {
+interface OwnProps {
     city: ExtendedCity
+    build(city: City, buildKey: string): void
 }
 
-export const CityInfo: React.SFC<TileInfoProps> = ({ city }) => (
+export const CityInfo: React.SFC<OwnProps> = ({ city, build }) => (
     <>
         <FlexContainer direction="column" grow={1} padding={5}>
             <p>City: {city.id}</p>
             <YieldInfo _yield={city.yield} />
+            Working on: <ProgressInfo progress={city.currentBuild || EMPTY_PROGRESS} />
         </FlexContainer>
         <FlexContainer direction="column" grow={1} padding={5}>
             Build:
+            {Object.entries(units).map(entry => (
```

Comment:
`Object.entries(units).map(([key, value]) => {` where you should rename `key` and `value` to describe what they are